sktime/forecasting/reservoir.py

```diff
@@ -0,0 +1,187 @@
+# -*- coding: utf-8 -*-
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Implements Next generation reservoir computing estimator."""
+
+__all__ = ["NextRC"]
+__author__ = ["frthjf"]
+
+import numpy as np
+import sklearn.linear_model
+from sklearn.utils.validation import check_array
+
+from sktime.forecasting.base import BaseForecaster
+
+
+class NextRC(BaseForecaster):
+    r"""Next generation reservoir based forecasts of time series data.
+
+    Uses a `sklearn` regressor `regressor` to regress values of time series on index:
+
+    In `fit`, for input time series :math:`(v_i, t_i), i = 1, \dots, T`,
+    where :math:`v_i` are values and :math:`t_i` are time stamps,
+    fits an `sklearn` model :math:`v_i = f(t_i) + \epsilon_i`, where `f` is
+    the model fitted when `regressor.fit` is passed `X` = vector of :math:`t_i`,
+    and `y` = vector of :math:`v_i`.
+
+    In `predict`, for a new time point :math:`t_*`, predicts :math:`f(t_*)`,
+    where :math:`f` is the function as fitted above in `fit`.
+
+    Default for `regressor` is linear regression = `sklearn` `Ridge` default.
+
+    If time stamps are `pd.DatetimeIndex`, fitted coefficients are in units
+    of days since start of 1970. If time stamps are `pd.PeriodIndex`,
+    coefficients are in units of (full) periods since start of 1970.
+
+    Parameters
+    ----------
+    regressor : estimator object, default = None
+        Define the regression model type. If not set, will default to
+         sklearn.linear_model.Ridge
+
+    Examples
+    --------
+    >>> from sktime.datasets import load_airline
+    >>> from sktime.forecasting.reservoir import NextRC
+    >>> y = load_airline()
+    >>> forecaster = NextRC()
+    >>> forecaster.fit(y)
+    NextRC(...)
+    >>> y_pred = forecaster.predict(fh=[1,2,3])
+    """
+
+    _tags = {
+        "y_inner_mtype": "pd.Series",
+        "X_inner_mtype": "pd.DataFrame",
+        "requires-fh-in-fit": False,
+        "handles-missing-data": True,
+        "scitype:y": "univariate",
+        "ignores-exogeneous-X": False,
```

Comment:
I made it `False` here, but still it ignores X data. LMK if I misunderstood here.

Reply:
if `True`, then we always pass `X=None` to the inner `_fit`, `_predict`. If `False`, then `X` can be `None` or non-empty. In either case, the inner `_fit` and `_predict` needs to be able to cope with `X=None` as per the current contract.